### PR TITLE
#184948256 Update password every x amount of time

### PR DESCRIPTION
**Pull request by Rwibutso**
#### What does this PR do?
This PR  prompt an interface to user to change password for x amount of timet
#### Description of Task to be completed?
-  Create a UI for the task
- Code the functionality
- Integrate with the backend
- Test the application
#### How should this be manually tested?
- clone this repo `git clone https://github.com/atlp-rwanda/e-comm-team-amigos-fn`
- switch your working directory `cd e-comm-team-amigos-fn`
- create a `.env` and update it with `.env-example`
- install all dependencies by running `npm install`
- in your terminal, run `npm start`
#### Test the endpoints:
-  `/update-password`
#### Any background context you want to provide?
- Node JS installed
- Node server running
#### What are the relevant pivotal tracker stories?
Story ID: [#184948256](https://www.pivotaltracker.com/story/show/184948256)
####Screenshots
Screenshot from 2023-04-28 12-14-22
Screenshot from 2023-04-28 12-14-34

Screenshot from 2023-05-02 14-19-09

### CODE REVIEW

**Comment by Emile-SHUMBUSHO** [comment]
This PR has a conflict to resolve

File: src/assets/css/changePassword.css [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+        margin-top: 0px;
+    }
+    .submitBtn{
+        height: 40px;
+        width: 80%;
+        padding: 7px;
+        margin-top: 0px;
+        font-size: 15px;
+    }
+}
```
Comment (Emile-SHUMBUSHO) [inline src/assets/css/changePassword.css]: Avoid an empty line from your code

File: src/components/changePasswordForm.jsx [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+            )}
+            <span className="errorDisplay" id="updateErrorDisplay"></span>
+            <input className='submitBtn' type='submit' value={'Update Password'} />
+          </form>
+        </div>
+      </div>
+    );
+  };
+  
+  export default ChangePasswordPage;
```
Comment (Emile-SHUMBUSHO) [inline src/components/changePasswordForm.jsx]: Avoid an empty line from your code

File: src/views/changePasswordPage.jsx [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import ChangePasswordForm from "../components/changePasswordForm.jsx";
+export default function ChangePasswordPage() {
+	return (
+		<>
+			<ChangePasswordForm />
+		</>
+	);
+}
```
Comment (Emile-SHUMBUSHO) [inline src/views/changePasswordPage.jsx]: Avoid an empty line from your code

File: src/components/changePasswordForm.jsx [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+              updateErrorDisplay.style.color = "red";
+            }else if (error.response.status == 404) {
+                updateErrorDisplay.innerHTML = "Email not found!";
+                updateErrorDisplay.style.color = "red";
+            }else {
+                console.log(error);
+            }
+        }
+    },
+  });
```
Comment (Emile-SHUMBUSHO) [inline src/components/changePasswordForm.jsx]: Great Job for making integration with API but I'm suggesting to separate Formik validation and integration functionality. Further more, take advantage of using redux to manage states  in your component.

**Review by Emile-SHUMBUSHO — CHANGES_REQUESTED** [review]

**Review by AbdulKhaliq59 — APPROVED** [review]
Rebase Unless it LGTM

**Review by Emile-SHUMBUSHO — APPROVED** [review]

**Review by khevin2 — APPROVED** [review]

File: src/views/updatePassword/updatePassword.style.scss [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+//         margin-top: 0px;
+//     }
+//     .submitBtn{
+//         height: 40px;
+//         width: 80%;
+//         padding: 7px;
+//         margin-top: 0px;
+//         font-size: 15px;
+//     }
+// }
```
Comment (Habinezajanvier) [inline src/views/updatePassword/updatePassword.style.scss]: Why is those commented codes, if they are not needed remove them at all

**Review by Habinezajanvier — CHANGES_REQUESTED** [review]

**Review by Habinezajanvier — APPROVED** [review]
👍

**Pull request merged**